evm/src/executor/abi.rs

```diff
@@ -680,6 +680,266 @@ pub static HARDHAT_CONSOLE_SELECTOR_PATCHES: Lazy<HashMap<Selector, Selector>> =
     ])
 });
 
+// Similar to hardhat's but ignores all specifiers except %s. %s works on all solidity values.
+// Support for other specifiers may be added in the future
+fn format_hardhat_log(spec: &str, v: &[&str]) -> String {
+    let mut result = spec.to_string().replace("%%", "%");
+
+    let mut end = v.len();
+    for (pos, t) in v.iter().enumerate() {
+        if result.find("%s").is_some() {
```

Comment:
You can use `contains("%s")` here instead

Reply:
we also need to decide what to do for things like:

`log("first: %s  second: %s", 10)`

how does hh handle this?
I'd assume it just replaces all `%s` with the value?

and

`log("first: %s  second: %s  third: %s", 10, 20)`

?